inc/sysrepo.h

```diff
@@ -128,6 +154,9 @@ typedef union sr_data_u {
  * @brief Structure that contains value of an data element stored in the sysrepo datastore.
  */
 typedef struct sr_val_s {
+    /** Memory context used by this value. */
+    sr_mem_ctx_t *sr_mem;
```

Comment:
I would name this as `_sr_mem` (with a `_` in the front) and explicitly state that this is an internal member (as in the .proto file)